src/fbrs/controller/AddNewUserController.java

```diff
@@ -1,24 +1,121 @@
 package fbrs.controller;
 
-import javafx.event.ActionEvent;
+import fbrs.model.DatabaseModel;
+import fbrs.model.Market;
+import fbrs.model.User;
 import javafx.fxml.Initializable;
-import javafx.scene.control.ComboBox;
-import javafx.scene.control.TextField;
+import javafx.geometry.NodeOrientation;
+import javafx.scene.control.*;
+import org.controlsfx.validation.Severity;
+import org.controlsfx.validation.ValidationResult;
+import org.controlsfx.validation.ValidationSupport;
+import org.controlsfx.validation.Validator;
 
 import java.net.URL;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 import java.util.ResourceBundle;
 
 public class AddNewUserController implements Initializable {
-    public TextField name;
-    public TextField phone;
-    public ComboBox type_user;
-    public ComboBox market;
+    private final Map<Integer, String> userTypeMap = new HashMap<>();
+    private DatabaseModel model = DatabaseModel.getModel();
+    private ValidationSupport support = new ValidationSupport();
 
-    public void onClickAdd(ActionEvent actionEvent) {
+    //UI
+    public TextField nameTextField;
+    public TextField phoneTextField;
+    public ComboBox<String> user_type;
+    public ComboBox<Market> marketComboBox;
+    public Label marketLabel;
+
+    public void onClickAdd() {
+        support.setErrorDecorationEnabled(true); // validate and show errors now!
+        Alert alert = new Alert(Alert.AlertType.ERROR);
+        alert.getDialogPane().setNodeOrientation(NodeOrientation.RIGHT_TO_LEFT);
+
+        int id = -1;
+        Market market = this.marketComboBox.getValue();
+
+        String message = "";
+
+        if (!support.isInvalid()) {
+            if (user_type.getValue().equals("تاجر")) {
+                if (market == null) {
+                    return;
+                }
+                id = model.addUser(market.getId(), this.nameTextField.getText(), this.phoneTextField.getText(),
+                        User.getUserIDTYPE(user_type.getValue()));
+
+                message = "تمت إضافة التاجر لسوق " + this.marketComboBox.getValue() + " بنجاح";
+
+            } else if (user_type.getValue().equals("صياد لنش") || user_type.getValue().equals("صياد حسكة")) {
+                id = model.addUser(-1, this.nameTextField.getText(), this.phoneTextField.getText(),
+                        User.getUserIDTYPE(user_type.getValue()));
+                message = "تمت إضافة " + user_type.getValue() + " بنجاح";
+            } else {
+                return;
+            }
+        }
+        if (id == -1) {
+            message = "فشلت علمية الإضافة";
+            alert.setHeaderText(message);
+            alert.setContentText("يرجى التأكد من إدخال البيانات بشكل صحيح وعدم إستخدام إسم مستخدم موجود بالفعل");
+            alert.showAndWait();
```

Comment:
You are repeating those lines in every if and else if, you can move them out instead